src/modules/storefront/put-storefront.ts

```diff
@@ -0,0 +1,71 @@
+import { notarize, Notarized, signPhantom } from '../notary';
+import { Solana } from '../solana/types';
+import { Storefront } from './types';
+
+export type PutStorefrontParams = Notarized<Storefront>;
+
+export interface PutStorefrontResult {
+  storefront: Storefront;
+}
+
+export const putStorefront = async ({
+  solana,
+  storefront,
+  onProgress,
+  onComplete,
+  onError,
+}: {
+  solana: Solana | undefined;
+  storefront: Storefront;
+  onProgress?: (
+    status: 'connecting-wallet' | 'signing' | 'uploading' | 'uploaded' | 'failed'
+  ) => void;
+  onComplete?: (result: PutStorefrontResult) => void;
+  onError?: (msg: string) => void;
+}): Promise<PutStorefrontResult> => {
+  try {
+    if (!onProgress) onProgress = () => {};
+
+    if (!solana) throw new Error('Could not connect to Solana');
+
+    if (!solana.isConnected) {
+      onProgress('connecting-wallet');
+
+      await Promise.all([new Promise<void>((ok) => solana.once('connect', ok)), solana.connect()]);
+    }
+
+    onProgress('signing');
+
+    const notarized: PutStorefrontParams = await notarize(storefront, signPhantom(solana));
+
+    onProgress('uploading');
+
+    const postResp = await fetch('/api/upload-store', {
+      method: 'POST',
+      body: JSON.stringify(notarized),
+    });
+
```

Comment:
```suggestion
    const postResp = await fetch('/api/storefronts', {
      method: 'PUT',
      body: JSON.stringify(notarized),
    });
```

How have you incorporated the onProgress? Can you share screenshot?

Reply:
I have not, the WIP commit contains a lot of missing glue code :sweat_smile: